server/hotrod/src/main/java/org/infinispan/server/hotrod/tx/table/Status.java

```diff
@@ -0,0 +1,84 @@
+package org.infinispan.server.hotrod.tx.table;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+
+import javax.transaction.TransactionManager;
+
+import org.infinispan.server.hotrod.tx.table.functions.TxFunction;
+
+/**
+ * Internal server status for the client's transactions.
+ * <p>
+ * The first 3 values are used as return value for the {@link TxFunction}.
+ *
+ * @author Pedro Ruivo
+ * @since 9.4
+ */
+public enum Status {
+   //used as return values
+   OK(0),
+   ERROR(1),
+   NO_TRANSACTION(2),
+
+   //real status
+   /**
+    * The client transaction is being replayed.
+    */
+   ACTIVE(3),
+   /**
+    * The client transaction was replayed successful (versions read are validated) and the transactions is preparing.
+    */
+   PREPARING(4),
+   /**
+    * The transaction is successful prepared and it waits for the client {@link TransactionManager} decision to commit
+    * or rollback
+    */
+   PREPARED(5),
+   /**
+    * The client {@link TransactionManager} decided to commit. It can't rollback anymore.
```

Comment:
What if the prepare fails?

Reply:
if the prepare fails, it changes from PREPARING to MARK_ROLLBACK and replies negatively to the transaction manager. The transaction manager has to decide to rollback the transaction.